Cache node_modules during deploy

chrismytton opened the pull request:
Now we're using timestamped deploys we want to avoid having to do a full `npm install` from scratch every time we deploy. This symlinks the `node_modules` directory into the vhost app directory before doing the npm install so modules are cached between deploys.

struan commented on config/post_deploy_actions.bash:
```diff
@@ -44,5 +44,10 @@ fi
+cache_dir="../cache"
+mkdir -p "$cache_dir/node_modules"
+ln -nfs "$cache_dir/node_modules" node_modules
+
```
I'd be a bit happier if this made sure it was actually in the correct directory in case something further up has done a cd.

struan commented:
Small comment but otherwise looks ok.

chrismytton commented:
I've changed the `post_deploy_actions.bash` script to use absolute paths everywhere, so there's no ambiguous paths being used. I've also switched to copying the whole node_modules directory into each release rather than symlinking it because this makes the concept of rolling back to a previous release as simple as switching the symlink to point to an older version.

It also accounts for the (very unlikely) situation where there is a node_module that gets dynamically required at some point during a deploy, which could be missing if it's removed in the newer release.
